File: from-books/recursion/language/python/exercise-3.py
# Write a function that accepts an array of strings and returns the total number of characters across all the strings.
# For example, if the input array is ["ab", "c", "def", "ghij"], the output should be 10 since there are 10 characters
# in total.
def numberOfChars(arr):
    if len(arr) == 1:
        return len(arr[0])

    sum = len(arr[0])

    return sum + numberOfChars(arr[1:])


def main():
    numberOfChars(["ab", "c", "def", "ghij"])


if __name__ == "__main__":
    main()

File: from-books/recursion/language/javascript/exercise-3.ts
/*
 * Write a function that accepts an array of strings and returns the total number of characters across all the strings.
 * For example, if the input array is ["ab", "c", "def", "ghij"], the output should be 10 since there are 10 characters
 * in total.
 */
function numberOfChars(arr: string[]): number {
  if (arr.length === 1) return arr[0].length;
  let sum = arr[0].length;
  return sum + numberOfChars(arr.slice(1, arr.length));
}

function main() {
  console.log(numberOfChars(["ab", "c", "def", "ghij"]));
}

main();
